lua/DCS-gRPC/methods/coalitions.lua

```diff
@@ -120,6 +120,70 @@ GRPC.methods.addGroup = function(params)
   return GRPC.success({group = GRPC.exporters.group(Group.getByName(template.name))})
 end
 
+GRPC.methods.addStaticObject = function(params)
+  if params.name == nil or params.name == "" then
+    return GRPC.errorInvalidArgument("name not supplied")
+  end
+  if params.type == nil or params.type == "" then
+    return GRPC.errorInvalidArgument("type not supplied")
+  end
+  if params.country_id == 0 or params.country_id == 15 then
+    return GRPC.errorInvalidArgument("invalid country code")
+  end
+
+  if params.offset == nil and params.position == nil then
+    return GRPC.errorInvalidArgument("provide either an offset or lat/lng position")
+  elseif params.offset ~= nil and (params.offset.unit == nil or params.offset.unit == "") then
+    return GRPC.errorInvalidArgument("provide the unit to ")
+  end
+
+  local staticTemplate = {
+    name = params.name,
+    type = params.type,
+    heading = math.rad(params.heading),
+    dead = params.dead,
+  }
+
+  if params.scoreRate ~= nil then
+    staticTemplate.rate = params.scoreRate
+  end
+
+  if params.cargoMass > 0 then
+    staticTemplate.canCargo = true
+    staticTemplate.mass = params.cargoMass
+  end
+
+  if params.offset ~= nil then
+    local unit = Unit.getByName(params.offset.unit)
+    if unit == nil then
+      return GRPC.errorNotFound("offset unit name not found")
+    end
+
+    -- we still need to supply a position to ED API, so current linked unit is good enough
+    local pos = unit:getPoint()
+    staticTemplate.x = pos.x
+    staticTemplate.y = pos.z
+
+    staticTemplate.linkOffset = true
+    staticTemplate.linkUnit = unit:getID()
+    staticTemplate.offsets = {
+      x = params.offset.x,
+      y = params.offset.y,
+      angle = math.rad(params.heading),
+    }
+  else
+    local pos = coord.LLtoLO(params.position.lat, params.position.lon)
+    staticTemplate.x = pos.x
+    staticTemplate.y = pos.z
+  end
+
+  coalition.addStaticObject(params.country - 1, staticTemplate)
+
+  return GRPC.success({
+    static = GRPC.exporters.static(StaticObject.getByName(params.name)),
```

Comment:
This may cause race condition issues. I have seen in the past that immediate retrieve calls in the same simulation frame can fail because the game hasn't fully instantiated the object yet.

It might be safer to not return anything. The `DCS-gRPC` client knows the details since it passed it all in apart from the ID and we tend not to use the object IDs in our APIs but use the names instead, which is known.

Reply:
You've got more experience with high-load servers, so I will lean on your experience and adjust. We may also want to adjust the following too then: https://github.com/DCS-gRPC/rust-server/blob/aa68abcb1d18cb0cc8c087362dbea9f5a2141c4b/lua/DCS-gRPC/methods/coalitions.lua#L120

PS: I agree with adding the fetch statics. It's next on my todo, I want to reduce the number of concurrent PR's.